image/helpers.go

```diff
@@ -73,7 +75,7 @@ func DownloadSnap(sto Store, name string, revision snap.Revision, opts *Download
 	targetFn = filepath.Join(targetDir, baseName)
 
 	pb := progress.NewTextProgress()
-	if err = sto.Download(name, targetFn, &snap.DownloadInfo, pb, opts.User); err != nil {
+	if err = sto.Download(nil, name, targetFn, &snap.DownloadInfo, pb, opts.User); err != nil {
```

Comment:
Instead of passing nil, I suggest using context.TODO() everywhere we lack such a value, so that we know a context parameter is always supposed to exist. Otherwise we'll have to do `if ctx != nil &&` everywhere, which is boring, and hopefully unnecessary since we'll gradually add it in the places we intend to support context.